src/pipx/commands/run.py

```diff
@@ -198,11 +198,11 @@ def run(
         # we can't parse this as a package
         package_name = app
 
-    content = None if spec is not None else maybe_script_content(app, is_path)
+    content = None if spec is not None else maybe_script_content(app, is_path)  # type: ignore[redundant-expr]
```

Comment:
`spec` is not `Optional`, so in theory it cannot be `None`.

Do we still want to defend against an actual `None`?